### ISSUE TITLE
Pytorch GPU Acceleration

### ISSUE THREAD

**Issue by IanChar** [uncertainty-toolbox/uncertainty-toolbox#81]
@tailintalent found that many of the evaluation metrics can be greatly sped up with gpu support via pytorch. This code is currently in an experimental branch called "torch". We will keep it separate from the main package for now. In order to be merged into main we will need the following:

* The code relying on torch needs to be separated out because we do not want to make torch a mandatory install. This needs to be done in the most robust way possible, ideally separating by pinpointing the exact operations where torch provides a speedup rather than making a torch version of the pre-existing code. 
* Set up an optional installation that allows users to select the torch-enabled version of the toolbox.
* Add options to the functions that allows users to select to use the torch version of the metric rather than the numpy version.

**Comment by tailintalent** [comment]
Sounds a good requirement. I think there are two relatively simple ways to satisfy the above requirement. One way is to add a global hyperparameter of BACKEND, which can be set as "numpy" or "torch". This BACKEND can be set by e.g. uct.set_backend("torch"). Then in each function with torch enhancement, this BACKEND will go into the "if" statement in each function to determine which input it expects. Another way is for the scripts with torch enhancement under uncertainty_toolbox/ folder, rename it as with a suffix of "_torch" and copy back the original script, and the BACKEND will choose which script to use.
